diff --git a/nav2_costmap_2d/include/nav2_costmap_2d/denoise/image_processing.hpp b/nav2_costmap_2d/include/nav2_costmap_2d/denoise/image_processing.hpp
--- a/nav2_costmap_2d/include/nav2_costmap_2d/denoise/image_processing.hpp
+++ b/nav2_costmap_2d/include/nav2_costmap_2d/denoise/image_processing.hpp
@@ -416,7 +416,8 @@ public:
     }

     // Label 0 is reserved for the background pixels, i.e. labels[0] is always 0
-    labels_ = {0};
+    labels_.clear();
+    labels_.resize(1, 0);
     next_free_ = 1;
   }
